packages/two_dimensional_scrollables/lib/src/common/implementation_span.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'span.dart';
+
+// This file is NOT exported as part of the package.
```

Comment:
I wonder if we could enforce this somehow with a lint...

Reply:
This is also violating the Flutter style, which requires everything to be exported (unless it is private with an underscore and hence not exportable): https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#structure

Why is it so important that this isn't exported? Could I use this as a building block for my own 2D scrollable?

Reply:
Oh I thought maybe that did not apply in flutter/packages? 😅  Since not everything in the style guide applies to all repos..

Regardless, I am more concerned about folks using this than the alternative of a little code duplication between the tree and the table, so I will put this back where it was as a private class.

 I don't want folks to be confused by the Span (+subclasses) they are meant to be using, and this one that is really just meant for internal use, especially as we would not want folks to use this and manipulate the information that should only be managed by the render object.